### ISSUE TITLE
Make time coordinates more user-friendly 

### ISSUE THREAD

**Issue by geek-yang** [AI4S2S/s2spy#34]
There is an issue with the current implementation of [calendar.resample()](https://github.com/AI4S2S/s2spy/blob/ceb894cc0527b6c712148f515857ede2c55c2628/s2s/time.py#L332-L334):

Related to xarray resample. 

- interval coordinate is now a Data variable. The datetime information is lost once you select the actual Data variable of interest (e.g. sst, or 1-d target timeseries). In this situation, you have the main index coordinate (unique integers), with multi-index anchor year and i_interval for your time indexing. The original datetime information is lost for the user.
- We had a discussion on the solution, and concluded that the interval Data variable should become a multi-index of the unique index coordinate. 

We need to keep the index coordinate unique and cannot replace the index coordinate by datetime labels, because datetime labels can conflict when going back in time more than 1 year.

**Comment by BSchilperoort** [comment]
We can easily set the intervals as a coordinate and not as a data variable by adding the following line to the resampling method:
```python
ds = ds.set_coords('interval')
```

As for the multi-index, I am not quite sure how we would implement that or if an actual pandas MultiIndex makes sense. In the way the dataset is currently created, it has a dimension `index`, which has multiple different coordinates to choose from.
These being;

- anchor year
- i_interval
- interval
- the actual unique **index**

So in a way, the current Dataset already has a "multi-index" (just in an xarray way)



### See the following Dataset:
![image](https://user-images.githubusercontent.com/12114825/176615515-089bf676-bfe8-4928-9d62-91d3d5f9e3dd.png)

#### Using the different coordinates:
The alternative coordinates can be used for operations such as `.groupby`:
```python
groups = ds_resampled.groupy('anchor_year')
```
The different coordinates can also set as the dimension, which allows operations such as `.sel` to work with them. E.g.:
```python
ds_resampled = ds_resampled.swap_dims({'index': 'anchor_year'})
ds_resampled.sel(anchor_year=2020)
```
```
>>> <xarray.Dataset>
>>> Dimensions:      (anchor_year: 3)
>>> Coordinates:
>>>     index        (anchor_year) int64 0 1 2
>>>   * anchor_year  (anchor_year) int32 2020 2020 2020
>>>     i_interval   (anchor_year) int64 0 1 2
>>>     interval     (anchor_year) object (2020-01-31, 2020-05-10] ... (2019-07-1...
>>> Data variables:
>>>     data         (anchor_year) float64 0.5297 0.2089 0.8391
```

**Comment by BSchilperoort** [comment]
> Can we make interval a 2d coordinate with dimensions anchor_year and i_interval?

Yes this is possible and would make some operations (such as selecting lags) more clean. It took me a while to figure out how to do this within xarray (and not relying on `np.reshape`), but the code below works: 
```python
# set anchor_year or i_interval as the main dimension (otherwise index is kept as dimension)
ds = ds.swap_dims({'index': 'anchor_year'}) 

# stack the anchor_year and i_interval into a multiindex
ds = ds.set_index(ai=('anchor_year', 'i_interval')) 

# unstack to make the data 2D with anchor_year and i_interval as dimensions
ds = ds.unstack() 
```

Returns:
```python
<xarray.Dataset>
Dimensions:      (anchor_year: 2, i_interval: 3)
Coordinates:
    index        (anchor_year, i_interval) int64 0 1 2 3 4 5
    interval     (anchor_year, i_interval) object (2020-01-31, 2020-05-10] .....
  * anchor_year  (anchor_year) int64 2020 2021
  * i_interval   (i_interval) int64 0 1 2
Data variables:
    data         (anchor_year, i_interval) float64 0.9239 0.5934 ... 0.4805
```
What do you think of this way of representing the data, @semvijverberg ?

**Comment by semvijverberg** [comment]
@Peter9192 Nice idea with the 2d coordinate. Getting used to it took about 1 minute, but it feels intuitive. It can indeed gives the lag selection and train-test splitting more clean feeling. From my side, I would say give this a go and we will see how it is to work with in practice.
